File: insertion.c
#include<stdio.h>
int i,temp,key,length,a[20];
void insert();
void sort();
void display();
void main(){
insert();
display();
sort();
display();
}


void insert(){
printf(" no of elements");
scanf("%d",&length);
for(i=0;i<length;i++){
scanf("%d",&a[i]);
}}
void sort(){
for(i=0;i<length;i++){
key=i;
while(a[key]>a[key+1] && key>=0 && key<4){
temp=a[key+1];
a[key+1]=a[key];
a[key]=temp;
key=key-1;
}
}}

void display(){
printf("the sorted array is");
for(i=0;i<length;i++){
printf("%d",a[i]);
}
}

File: insertion.py

a=[]
length=int(input("no of elements"))
def insert():
    global length
    for i in range(0,length):
        element=int(input(" "))
        a.append(element)

def sort():
    global length
    for j in range(0,length):
        key=j
        while key>0 and a[key]<a[key-1]:
            temp=a[key+1]
            a[key+1]=a[key]
            a[key]=temp
            key=key-1
    
        
def display():
    global length
    for i in range(0,length):
        print(a[i])


insert()
sort()
display()

            
        
